elastic/beats
# Fix rds metricset from reporting the same value for different instances

rds metricset makes AWS API calls to cloudwatch to get rds cloudwatch metrics. Some metrics are correct but some of the metrics are with wrong values.

This bug is found in discuss forum:
https://discuss.elastic.co/t/rds-module-only-getting-metrics-of-one-instance

### Problem
In the same region, rds metricset is collecting metrics from each instance and overwriting one set of metrics with values from another. rds arn should not be assigned to each metric because some cloudwatch metrics of rds has dimensions like database class and database engine.

With bug, you would not see difference between two rds instance metrics like the picture below. Two lines would actually be overlapping.

### Fix
Remove arn from labels for each cloudwatch metric, only use the dimensions from each metric instead.

### Result to prove it's fixed
With this fix, rds metricset is reporting correct values.
Diagram from Cloudwatch:
<img width="731" alt="Screen Shot 2019-11-22 at 1 30 52 PM" src="https://user-images.githubusercontent.com/14081635/69458384-6a380600-0d2c-11ea-9c37-76eb448e10e3.png">

Diagram from aws rds metricset:
<img width="1618" alt="Screen Shot 2019-11-22 at 1 31 03 PM" src="https://user-images.githubusercontent.com/14081635/69458378-660be880-0d2c-11ea-8d7b-822bc7b33d17.png">

Also in this PR, I changed `cpu.total.pct` to be the (actual value/100) in order to make `format: percent` work out of the box in Kibana metric explorer. Without the fix, it will show 400% in Kibana instead of 4%.

## How to Test this:
Build metricbeat with this PR
Enable aws module
Edit modules.d/aws.yml with
```
- module: aws
  period: 60s
  credential_profile_name: test-mb
  metricsets:
    - rds
```
After several minutes, check the metrics. Some metrics will have the same value but should not always be the same. Compare with AWS Cloudwatch portal with metrics values collected from aws module. Metrics from metricbeat will be a little bit delayed but should have the same values with average statistics.
